aquia-inc/template-vite-react
# build(dev-deps): update typescript and types

## Summary

Upgraded TypeScript and several related development dependencies to their latest versions to ensure compatibility and improved performance.

### Added

- None

### Changed

- Upgraded `@types/lodash` from `^4.17.4` to `^4.17.7`
- Upgraded `@types/node` from `^20.12.12` to `^20.14.14`
- Upgraded `@typescript-eslint/eslint-plugin` from `^7.16.1` to `^7.18.0`
- Upgraded `@typescript-eslint/parser` from `^7.10.0` to `^7.18.0`
- Upgraded `typescript` from `^5.4.5` to `^5.5.4`

### Deprecated

- None

### Removed

- None

### Fixed

- None

## How to test

1. Checkout this branch: `git checkout origin/build/dependencies/5-typescript-types`
2. Run `yarn` and ensure all devDependencies are correctly installed without duplication and that there are no issues related to missing or incompatible dependencies.
3. Run `yarn test` and verify that all tests run and pass.
4. Run `yarn dev` and verify that the application runs as expected.
5. Run `yarn build` and verify that the application builds as expected.
6. Run `yarn build-storybook` and verify the Storybook build succeeds.
